src/lib/Component/Renderer/DefaultRenderer.php

```diff
@@ -57,15 +57,28 @@ public function renderGroup(string $groupName, array $parameters = []): array
      */
     public function renderSingle(string $groupName, string $name, array $parameters = []): string
     {
+        $components = $this->registry->getComponents($groupName);
+
+        return $this->renderSingleWithComponents($groupName, $name, $components, $parameters);
+    }
+
+    /**
+     * @param array<mixed> $parameters
+     * @param array<string, \Ibexa\Contracts\TwigComponents\ComponentInterface> $components
+     */
+    private function renderSingleWithComponents(
```

Comment:
I would personally name it `doRenderSingle`, but there is not really an established pattern in the team and it's debatable. Will be good as it is, unless someone else with stronger opinion wants to chime in.

:+1: